compiler/expressions.go

```diff
@@ -474,10 +474,23 @@ func (fc *funcContext) translateExpr(expr ast.Expr) *expression {
 				fc.pkgCtx.errList = append(fc.pkgCtx.errList, types.Error{Fset: fc.pkgCtx.fileSet, Pos: e.Index.Pos(), Msg: "cannot use js.Object as map key"})
 			}
 			key := fmt.Sprintf("%s.keyFor(%s)", fc.typeName(t.Key()), fc.translateImplicitConversion(e.Index, t.Key()))
+			entryVar := fc.newVariable("_entry")
 			if _, isTuple := exprType.(*types.Tuple); isTuple {
-				return fc.formatExpr(`(%1s = %2e[%3s], %1s !== undefined ? [%1s.v, true] : [%4e, false])`, fc.newVariable("_entry"), e.X, key, fc.zeroValue(t.Elem()))
-			}
-			return fc.formatExpr(`(%1s = %2e[%3s], %1s !== undefined ? %1s.v : %4e)`, fc.newVariable("_entry"), e.X, key, fc.zeroValue(t.Elem()))
+				return fc.formatExpr(
+					`(%1s = typeof %2e.get === "function" ? %2e.get(%3s) : undefined, %1s !== undefined ? [%1s, true] : [%4e, false])`,
+					entryVar,
+					e.X,
+					key,
+					fc.zeroValue(t.Elem()),
+				)
+			}
```

Comment:
Sorry if this goes against preferred formatting.  My brain was unable to parse the long line and multiple arguments.  I can put it back to single line, if preferred.

Reply:
My 2¢: This seems fine, and more readable than before.

Reply:
I have no objections either :)